src/flow-control/input-handler.js

```diff
@@ -36,4 +39,11 @@ module.exports = class InputHandler {
 
         return commands;
     }
+
+    onFinish() {
+        if (this.inputStream && this.pauseInputStreamOnFinish) {
+            // https://github.com/kimmobrunfeldt/concurrently/issues/252
+            this.inputStream.pause();
```

Comment:
Rather than pausing, WDYT if this called `.unref()` instead?
We can then get rid of one the `pauseInputStreamOnFinish` option.

Reply:
`unref` is only a function on `net.Socket`, but not readable Stream, so it doesn't work if `stdin` is a file.
```
$ concurrently ... < foo
(node:51227) UnhandledPromiseRejectionWarning: TypeError: process.stdin.unref is not a function
...
```
We could update the docs to say "`inputStream` is either `net.Socket` or readable Stream" and then have this clean up check if the `unref` function exists and calls that if it does, but I think that adds too much complexity. Thoughts?

Reply:
Ah yes, of course it couldn't be this simple 🤦 OK. Need to think a bit more.

Reply:
Hmm, I think we could be smart and pause the input stream if it has no more listeners and it either
- was paused before; or
- was not flowing before.

Although, since I haven't been able to give much attention to concurrently, I'd say just introducing the option is probably best.

WDYT?

Reply:
It's been so long since I worked on this PR, honestly, I can't provide any more insights. (it seems like you are similarly distracted as well) If you're suggesting merging the PR as-is, that would be preferable by me. Are there any more blocking concerns you'd like me to address first?

Reply:
Just the handler interface, as per https://github.com/open-cli-tools/concurrently/pull/253#discussion_r637614525